src/daemon/start_mon.sh

```diff
@@ -192,7 +192,11 @@ function start_mon {
     # DO NOT TOUCH IT, IT MUST BE PRESENT
     DAEMON_OPTS+=(--mon-cluster-log-to-stderr "--log-stderr-prefix=debug ")
     log "SUCCESS"
-    exec /usr/bin/ceph-mon "${DAEMON_OPTS[@]}" -i "${MON_NAME}" --mon-data "$MON_DATA_DIR" --public-addr "${MON_IP}"
+    # Set PUBLIC_BIND_ADDRESS to MON_IP if unset
+    if [ -z "${PUBLIC_BIND_ADDRESS}" ]; then
+        PUBLIC_BIND_ADDRESS=${MON_IP}
+    fi
+    exec /usr/bin/ceph-mon "${DAEMON_OPTS[@]}" -i "${MON_NAME}" --mon-data "$MON_DATA_DIR" --public-addr "${PUBLIC_BIND_ADDRESS}" --public-bind-addr "${MON_IP}"
```

Comment:
This is a little bit confusing because you're using the PUBLIC_BIND_ADDRESS variable for --public-addr option.
It would be better to rename the variable to something like MON_PUBLIC_ADDRESS and keep MON_IP for --public-bind-addr.
What do you think ?

Reply:
You are right, this is a little bit confusing. If the "--public-bind-address" is used, the monitor tries to bind the network interface which is provided by this parameter. The address passed by "--public-addr" is used for external authentication (https://github.com/ceph/ceph-client/blob/master/net/ceph/messenger.c#L2020).